core/app/models/spree/line_item.rb

```diff
@@ -88,12 +87,11 @@ def pre_tax_amount
     alias display_total display_amount
     deprecate display_total: :display_amount, deprecator: Spree::Deprecation
 
-    # Sets price and currency from a `Spree::Money` object
+    # Sets price from a `Spree::Money` object
     #
-    # @param [Spree::Money] money - the money object to obtain price and currency from
+    # @param [Spree::Money] money - the money object to obtain price from
     def money_price=(money)
```

Comment:
I think this should have a Safeguard: It should raise a good error when trying to assign a price with a currency different to the order currency. Otherwise the information would just be lost. So something along these lines: 

``` ruby
class CurrencyMismatch < StandardError; end

def money_price=(money)
   return unless money
   raise CurrencyMismatch, "Line item price currency must match order currency!" if money.currency.iso_code != currency
   self.price = money.to_d
end